DOCS-6851-added new content (#18)
mulesoft/docs-service-mesh: modules/ROOT/pages/create-an-api-configure-service-mesh-CLI.adoc

= Create an API Using the CLI
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: service mesh, microservices, downloading, installing

> After you provision the adapter, you must create an API to bind to the service. The following diagram explains the flow for creating an API:

Reviewer: ```suggestion
After you provision the adapter, you must create an API to bind to the service. The following diagram illustrates the flow for creating an API:
```

Author: Done! Thanks!

//image::configure-service-mesh-flowchart-2.png[]
image::create-API-CLI-flowchart.png[]

You can create an API using Anypoint Platform or Anypoint Service Mesh:

* <<create-api-AP, Create an API using Anypoint Platform>>
+
If you want to create REST or SOAP APIs with OAS or WSDL definition, use Anypoint Platform. You can manage your contracts and complex API configurations easily with Anypoint Platform.
+
Note that if you want to upload a REST API or OAS asset type, you upload a .json or .yaml file. Alternatively, if you want to upload a SOAP API or WSDL asset type, you upload a .wsdl or .xml file.

> * <<create-api-ASM-CLI, Create an API using Anypoint Service Mesh Autocreate>>

Reviewer 2: @priyaayyar This header looks really odd. Please change it to 

> Create an API using Anypoint Service Mesh (Auto-create)

Reviewer: If you do this, suggest Auto-Create an API Using....

Author: Went with Create an API using Anypoint Service Mesh

+

> The Autocreate feature of Anypoint Service Mesh allows you to create an HTTP API with no contracts or definitions. When you choose this option, Anypoint Service Mesh creates the API for you in Anypoint Platform. You do not have to manually log in to Anypoint Platform. 

Reviewer 2: > The auto-create feature

Reviewer: ```suggestion
The auto-create feature of Anypoint Service Mesh enables you to create an HTTP API with no contracts or definitions. When you choose this option, Anypoint Service Mesh creates the API for you in Anypoint Platform. You do not have to manually log in to Anypoint Platform. 
```

Author: Done! Thanks!

+
Note that if you create REST or SOAP APIs using the auto-create feature of Anypoint Service Mesh, the APIs are still managed internally as an HTTP asset without any definition. 
+

> *Tomas comment: IMHO, It should talk about contracts here. You can use contracts even if you created your API with SM.*

Reviewer: Resolved?

Author: Done! Thanks!

[[create-api-AP]]
== Creating an API Using Anypoint Platform

To create an API using Anypoint Platform, perform the following steps:

. Go to *Anypoint Platform > API Manager* and select the *Manage API* drop-down list. 
. Select *Create new API*.
+

> The Creating an asset page is displayed.

Reviewer: Is the page labeled this way, in sentence case?

Author: Done! Thanks!

> . Specify the *Name*, and *Asset types* details for the API.  

Reviewer: ```suggestion
. Specify a *Name* value and *Asset types* details for the API.  
```

Author: Done! Thanks!

+

> Your Asset is now created in Exchange for reuse. After you create the Asset, you are now ready to perform Endpoint configuration.

Reviewer: ```suggestion
Your asset is now created in Exchange for reuse. After you create the asset, you are ready to perform endpoint configuration.
```

Author: Done! Thanks!

+
The Endpoint Configuration window is displayed.
. In the *Managing type* field, select the *Basic Endpoint* radio button.
. In the *Application type* field, select the *Non-Mule application* radio button.
. Optionally, specify the *Implementation URI*, and click *Save*.
+
Your API is now created.

[[create-api-ASM-CLI]]
== Creating an API Using Anypoint Service Mesh

Before you create the API, you must complete the task prerequisite.

*Task Prerequisite*

> Ensure that you obtain the environment ID and organization ID. You can perform this task from either API Manager or from Access Management. You require the <INSERT_PERMISSION> permission or <INSERT_ROLE> role to see to obtain these credentials.

Reviewer: Do you have the values for these?

Author: Done! Thanks!

> *Tomas comment: In order to create an API you will need to have at least Exchange Contributor for the Business group where you are creating the asset and Manage APIs configuration for the environment where you are creating the API Instance. But keep in mind that that is different from the permission Ensure that you obtain the environment ID and organization ID. For the latter permission is just access to APIM.*

Reviewer: Resolved?

Author: Done! Thanks!

To create an API using Anypoint Service Mesh, perform the following steps:

> . Using the Anypoint Service Mesh, type the following command to create the API:

Reviewer: ```suggestion
. Using Anypoint Service Mesh, type the following command to create the API:
```

Author: Done! Thanks!

+
[source,text,linenums]
----
./service-mesh api create  
--name=<API resource name> \
--namespace=<namespace name> \
--environmentId=<environment ID> \
--groupId=<group ID of the asset, for example, organization ID> \
--assetId=<asset ID on Exchange> \
--version=<Asset version of the instance on exchange> \
--instanceLabel=<ID of the instance on API Manager> \ 
--specTags=<list of comma separated tags to be added to the asset on Exchange> \
--instanceTags=<list of comma separated tags to be added to the instance on API Manager> \
--user=<username of the Anypoint Platform user, creating the API instance> \
--password=<password of the Anypoint Platform user, creating the API instance>
----
+
[%header%autowidth.spread]
|===
| Parameter Name | Description | Required/Optional | Default Value

> | `name` | The name of the API that you want to provision, for example, <orders-api>. | Required | N/A

Reviewer: GLOBALLY review and standardize formats used for values in this table column throughout: < >, ", `....

Author: Done! Thanks!

| `namespace` | The namespace in which you want to create the adapter, for example, “mulesoft-apps”. | Optional | default
| `environmentId` | The ID of the environment where you want to create your API instance. | Required | N/A
| `groupId` | The ID of the group of assets on Exchange, for example, the organization ID. You can find the organization ID on API Manager landing page or in Access Management. | Optional | Organization ID of the organization that contains the environment. 
| `assetId` | The ID of the assets that are a part of an organization. Supports alphanumeric characters without spaces. | Required | N/A

> | `version` | The version of the asset on Exchange. *Needs to be semver* what is this chirag? | Required | N/A

Reviewer: Is this resolved?

Author: Done! Thanks!

| `instanceLabel` | The ID of the API instances of an Asset within a given environment. Supports alphanumeric characters without spaces. | Required | N/A
| `specTags` | Tags to be added to the asset on Exchange. |  Optional | N/A
| `instanceTags` | Tags to be added to the API instance on API Manager. |  Optional | N/A

> | `user` | The username credential of the Anypoint Platform user, who is creating the API. | Required | N/A

Reviewer: ```suggestion
| `user` | The username credential of the Anypoint Platform user who is creating the API. | Required | N/A
```

Author: Done! Thanks!

> | `password` | The password of the Anypoint Platform user, who is creating the API. | Required | N/A

Reviewer: ```suggestion
| `password` | The password of the Anypoint Platform user who is creating the API. | Required | N/A
```

Author: Done! Thanks!

|===
+
. Verify the <<provision-adapter-configure-service-mesh-CLI#adapter_status, adapter status>>:
+

> `./service-mesh adapter list`

Reviewer 2: @priyaayyar @tbruno Why is the adapter list section under api creation section? This is wrong and it needs to change to api list with this command 

>  ./service-mesh api list

Reviewer: Resolved?

Author: Done! Thanks!

+

> The adapter displays a Ready status.

Reviewer 2: Please change this to > The API displays a Ready status.

Author: Done! Thanks!
